src/ApacheOrcDotNet.OptimizedReader/Buffers/BaseColumnBuffer.cs

```diff
@@ -55,100 +44,89 @@ private protected void ReadByteStream(StreamDetail stream, ReadOnlySpan<byte> de
-                        return;
+                    if (isFirstByte)
+                    {
+                        isFirstByte = false;
 
-                    outputValues[numValuesRead++] = (decodedByte & 32) != 0;
-                    if (checkRemainingBits && ++numSkippedBits <= stream.Positions.RemainingBits)
-                        numValuesRead--;
-                    if (numValuesRead >= outputValues.Length)
-                        return;
+                        var firstValues = rleValues[stream.Positions.RemainingBits..];
 
-                    outputValues[numValuesRead++] = (decodedByte & 16) != 0;
-                    if (checkRemainingBits && ++numSkippedBits <= stream.Positions.RemainingBits)
-                        numValuesRead--;
-                    if (numValuesRead >= outputValues.Length)
-                        return;
+                        numValuesRead += firstValues.Length;
 
-                    outputValues[numValuesRead++] = (decodedByte & 8) != 0;
-                    if (checkRemainingBits && ++numSkippedBits <= stream.Positions.RemainingBits)
-                        numValuesRead--;
-                    if (numValuesRead >= outputValues.Length)
-                        return;
-
-                    outputValues[numValuesRead++] = (decodedByte & 4) != 0;
-                    if (checkRemainingBits && ++numSkippedBits <= stream.Positions.RemainingBits)
-                        numValuesRead--;
-                    if (numValuesRead >= outputValues.Length)
-                        return;
+                        firstValues.CopyTo(outputValues);
+                    }
+                    else
+                    {
+                        var targetBuffer = outputValues[numValuesRead..];
 
-                    outputValues[numValuesRead++] = (decodedByte & 2) != 0;
-                    if (checkRemainingBits && ++numSkippedBits <= stream.Positions.RemainingBits)
-                        numValuesRead--;
-                    if (numValuesRead >= outputValues.Length)
-                        return;
+                        if (numValuesRead + rleValues.Length >= outputValues.Length)
+                        {
+                            var source = rleValues[..(outputValues.Length - numValuesRead)];
+                            numValuesRead += source.Length;
+                            source.CopyTo(targetBuffer);
+                            return;
+                        }
 
-                    outputValues[numValuesRead++] = (decodedByte & 1) != 0;
-                    if (checkRemainingBits && ++numSkippedBits <= stream.Positions.RemainingBits)
-                        return;
-                    if (numValuesRead >= outputValues.Length)
-                        return;
+                        numValuesRead += rleValues.Length;
 
-                    if (checkRemainingBits)
-                        checkRemainingBits = false;
+                        rleValues.CopyTo(targetBuffer);
```

Comment:
How did this impact performance vs the previous route?  It's certainly easier on the eyes.

Reply:
It did not add any performance impact and the execution times stayed in the same average values after the change.